Updates to unix build script

codito opened the pull request:
* Allow build of entire TestPlatform.sln if mono is installed
* Initial version of test script
* Use cross platform enabled nunit framework

Commit:
Updates to build.sh.
Allow building entire solution if mono is available on the box.
Initial version of test.sh script.

Commit:
Update version to 15.1.0 for build.

msftclas commented:
@codito,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

Commit:
Merge remote-tracking branch 'upstream/master' into build-sh

Commit:
Fix unbound variable for packages directory.

Commit:
Merge remote-tracking branch 'upstream/master' into build-sh

Commit:
Fix condition for checking presence of mono.
Package TestHost in CLI nuget package if mono is available and net46 binaries are built.
Fix an issue with CLI nupkg, where an empty folder was getting created in the nupkg.

Commit:
Override json.net in coreclr package directory.

codito committed:
Merge branch 'master' into build-sh

Commit:
Download netcoreapp1.0 shared runtime.

codito commented:
@harshjain2 can you take a look please?

codito committed:
Merge branch 'master' into build-sh

harshjain2 commented on src/package/package/package.csproj:
```diff
@@ -43,7 +43,7 @@
     <ProjectReference Include="$(TestPlatformRoot)src\Microsoft.TestPlatform.VsTestConsole.TranslationLayer\Microsoft.TestPlatform.VsTestConsole.TranslationLayer.csproj" />
     <ProjectReference Include="$(TestPlatformRoot)src\Microsoft.TestPlatform.Extensions.TrxLogger\Microsoft.TestPlatform.Extensions.TrxLogger.csproj" />
-	<ProjectReference Include="$(TestPlatformRoot)src\Microsoft.TestPlatform.TestHostProvider\Microsoft.TestPlatform.TestHostProvider.csproj" />
+    <ProjectReference Include="$(TestPlatformRoot)src\Microsoft.TestPlatform.TestHostProvider\Microsoft.TestPlatform.TestHostProvider.csproj" />
```
nit : whitespaces.

harshjain2 commented on build.sh:
```diff
@@ -1,9 +1,11 @@
 
 if [[ $? -ne 0 ]]; then
-	exit 1
+    exit 1
```
nit : formatting.

codito replied:
Modified code converted the tabs to spaces :)
